File: backend/Models/Quizz.cs
using System.ComponentModel.DataAnnotations;

namespace prid_2324_a02.Models;

public class Quizz
{
	[Key]
	public int Id { get; set; }
	public string Name { get; set; } = null!;
	public string? Description { get; set; }
	public int DatabaseId { get; set; }
	public Database Database { get; set; } = null!;
	public bool IsPublished { get; set; }
	public bool IsClosed { get; set; } = false;
	public bool IsTest { get; set; } = false;
	public DateTimeOffset? Start { get; set; }
	public DateTimeOffset? Finish { get; set; }

	public virtual ICollection<Attempt> Attempts { get; set; } = new HashSet<Attempt>();
	public virtual ICollection<Question> Questions { get; set; } = new HashSet<Question>();
}

File: frontend/src/app/models/quizz.ts
import { Type } from "class-transformer";
import 'reflect-metadata';
import { Database } from "./database";
import { Question } from "./question";

export class Quizz {
	id?: number;
	name?: string;
	description?: string;
	isPublished?: boolean;
	isClosed?: boolean;
	isTest?: boolean;
	@Type(() => Date)
	start?: Date;
	@Type(() => Date)
	finish?: Date;
	database?: Database;
	status?: string;
	editable?: boolean;

	questions: Question[] = [];

	get display(): string {
		return `${this.name} - (${this.description ? this.description : 'no description'})`;
	}

	get evaluation(): string {
		return "N/A";
	}

	get getTestOrTraining(): string {
		return this.isTest ? "Test" : "Training";
	}

	get dbName(): string | undefined {
		return this.database?.name;
	}

	get startAsString(): string {
		return this.start ? this.start.toLocaleDateString('eu-EU') : "N/A";
	}

	get finishAsString(): string {
		return this.finish ? this.finish.toLocaleDateString('eu-EU') : "N/A";
	}
}
